include/aws/common/atomics_gnu_old.inl

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/* These are implicitly included, but help with editor highlighting */
+#include <aws/common/atomics.h>
+#include <aws/common/common.h>
+
+#include <stdint.h>
+#include <stdlib.h>
+
+struct aws_atomic_var {
+    union {
+        size_t intval;
+        void *ptrval;
+    } u;
+};
+
+#define AWS_ATOMIC_INIT_INT_IMPL(x)                                                                                    \
+    {                                                                                                                  \
+        .u = {.intval = (x) }                                                                                          \
+    }
+#define AWS_ATOMIC_INIT_PTR_IMPL(x)                                                                                    \
+    {                                                                                                                  \
+        .u = {.ptrval = (void *)(x) }                                                                                  \
+    }
+
+static inline void aws_atomic_private_compiler_barrier() {
+    __asm__ __volatile__("" : : : "memory");
```

Comment:
is this going to work anywhere other than intel?

Reply:
I believe this will work with all GCC-(more-or-less) compatible compilers, it's not tied to the architecture.